src/goto-symex/shadow_memory_util.cpp

```diff
@@ -71,3 +78,108 @@ void clean_pointer_expr(exprt &expr, const typet &type)
+  const exprt &expr)
+{
+  log.conditional_output(
+    log.debug(), [ns, field_name, expr](messaget::mstreamt &mstream) {
+      mstream << "Shadow memory: get_field: " << id2string(field_name)
+              << " for " << format(expr) << messaget::eom;
+    });
+}
+
+void log_value_set(
+  const namespacet &ns,
+  const messaget &log,
+  const std::vector<exprt> &value_set)
+{
+#ifdef DEBUG_SM
+  log.conditional_output(
+    log.debug(), [ns, value_set](messaget::mstreamt &mstream) {
+      for(const auto &e : value_set)
+      {
+        mstream << "Shadow memory: value set: " << format(e) << messaget::eom;
+      }
+    });
+#endif
+}
+
+void log_value_set_match(
+  const namespacet &ns,
+  const messaget &log,
+  const shadow_memory_statet::shadowed_addresst &shadowed_address,
+  const exprt &matched_base_address,
+  const value_set_dereferencet::valuet &dereference,
+  const exprt &expr,
+  const value_set_dereferencet::valuet &shadow_dereference)
+{
+#ifdef DEBUG_SM
+  log.conditional_output(
+    log.debug(),
+    [ns,
+     shadowed_address,
+     expr,
+     dereference,
+     matched_base_address,
+     shadow_dereference](messaget::mstreamt &mstream) {
+      mstream << "Shadow memory: value set match: " << messaget::eom;
+      mstream << "Shadow memory:   base: " << format(shadowed_address.address)
+              << " <-- " << format(matched_base_address) << messaget::eom;
+      mstream << "Shadow memory:   cell: " << format(dereference.pointer)
+              << " <-- " << format(expr) << messaget::eom;
+      mstream << "Shadow memory:   shadow ptr: "
+              << format(shadow_dereference.pointer) << messaget::eom;
+      mstream << "Shadow memory:   shadow val: "
+              << format(shadow_dereference.value) << messaget::eom;
+    });
+#endif
+}
+
+void log_value_set_match(
+  const namespacet &ns,
+  const messaget &log,
+  const exprt &address,
+  const exprt &expr)
+{
+  // Leave guards rename to DEBUG_SHADOW_MEMORY
+#ifdef DEBUG_SM
+  log.conditional_output(
+    log.debug(), [ns, address, expr](messaget::mstreamt &mstream) {
+      mstream << "Shadow memory: value set match: " << format(address)
+              << " <-- " << format(expr) << messaget::eom;
+    });
+#endif
+}
+
+void log_try_shadow_address(
+  const namespacet &ns,
+  const messaget &log,
+  const shadow_memory_statet::shadowed_addresst &shadowed_address)
+{
+#ifdef DEBUG_SM
+  log.conditional_output(
+    log.debug(), [ns, shadowed_address](messaget::mstreamt &mstream) {
+      mstream << "Shadow memory: trying shadowed address: "
+              << format(shadowed_address.address) << messaget::eom;
+    });
+#endif
+}
+
+void log_cond(
+  const namespacet &ns,
+  const messaget &log,
+  const char *cond_text,
+  const exprt &cond)
+{
+#ifdef DEBUG_SM
+  log.conditional_output(
+    log.debug(), [ns, cond_text, cond](messaget::mstreamt &mstream) {
+      mstream << "Shadow memory: " << cond_text << ": " << format(cond)
```

Comment:
⛏️  missing `cond: ` in message

Reply:
`cond_text` that is logged just before the `:` is the condition

Reply:
The condition is in `cond_text` before the `:`